src/encoded/static/libs/ui/node_graph.js

```diff
@@ -0,0 +1,254 @@
+        text.each(function textWrap() {
+            text = d3.select(this);
+            const words = text.text().split(/\s+/);
+            let line = [];
+            let lineNumber = 0;
+            const lineHeight = 1.1;
+            const y = text.attr('y');
+            const dy = parseFloat(text.attr('dy'));
+            let newDy = '';
+            let tspan = text.text(null).append('tspan').attr('x', 0).attr('y', y);
+            tspan = text.append('tspan').attr('x', 0).attr('y', y);
+            words.forEach((word) => {
+                line.push(word);
+                tspan.text(line.join(' '));
+                if (tspan.node().getComputedTextLength() > textWidth) {
+                    line.pop();
+                    tspan.text(line.join(' '));
+                    line = [word];
+                    tspan = text.append('tspan').attr('x', 0).attr('y', y);
+                    newDy = `${(lineNumber += 1) * lineHeight + dy}em`;
+                    tspan.attr('dy', newDy);
+                    tspan.text(word);
+                }
+            });
+        });
+    }
+
+    function update(source, animationDuration) {
+        function click(event, d) {
+            if (d.children) {
+                d._children = d.children;
+                d.children = null;
+            } else {
+                d.children = d._children;
+                d._children = null;
+            }
+            update(d, slowAnimation);
+        }
+
+        const treeData = treemap(root);
+        const nodes = treeData.descendants();
+        const links = treeData.descendants().slice(1);
+        const nodeDepth = nodes.map((n) => n.depth);
+        const maxNodeDepth = Math.max(...nodeDepth);
+
+        nodes.forEach((d) => {
+            d.id = nodeKeyName(d.data.name);
+            if (d.depth === 0) {
+                d.y = d.depth * 100;
+            } else {
+                d.y = (d.depth * height) / maxNodeDepth;
+            }
+        });
+
+        const node = svg.selectAll('g.node')
+            .data(nodes, (d) => d.id);
+
+        const nodeEnter = node.enter().append('g')
+            .attr('class', (d) => `node ${d.data.class} ${d.data.name}`)
+            .attr('transform', `translate(${source.x0},${source.y0})`)
+            .on('click', (e, d) => {
+                setSelectedNodes(d.data.name);
+                const clickedName = nodeKeyName(d.data.name);
+                if (internalSelectedNodes.indexOf(clickedName) > -1 && internalSelectedNodes.length > 1) {
+                    internalSelectedNodes = internalSelectedNodes.filter((s) => s !== clickedName);
+                } else if (internalSelectedNodes.indexOf(clickedName) > -1) {
+                    internalSelectedNodes = [];
+                } else {
+                    internalSelectedNodes = [...internalSelectedNodes, clickedName];
+                }
+                const clickedCell = d3.select(`.js-cell-${nodeKeyName(d.data.name)}`);
```

Comment:
Are you too a fan of prefixing css not meant for styling but for Javascript to identify specific DOM nodes, with 'js-'?

Reply:
I think I copied this from the original ticket